infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerMachineStarter.java

```diff
@@ -0,0 +1,774 @@
+import org.eclipse.che.plugin.docker.client.json.Volume;
+import org.eclipse.che.plugin.docker.client.json.container.NetworkingConfig;
+import org.eclipse.che.plugin.docker.client.json.network.ConnectContainer;
+import org.eclipse.che.plugin.docker.client.json.network.EndpointConfig;
+import org.eclipse.che.plugin.docker.client.params.BuildImageParams;
+import org.eclipse.che.plugin.docker.client.params.CreateContainerParams;
+import org.eclipse.che.plugin.docker.client.params.ListImagesParams;
+import org.eclipse.che.plugin.docker.client.params.PullParams;
+import org.eclipse.che.plugin.docker.client.params.RemoveContainerParams;
+import org.eclipse.che.plugin.docker.client.params.RemoveImageParams;
+import org.eclipse.che.plugin.docker.client.params.StartContainerParams;
+import org.eclipse.che.plugin.docker.client.params.TagParams;
+import org.eclipse.che.plugin.docker.client.params.network.ConnectContainerToNetworkParams;
+import org.eclipse.che.workspace.infrastructure.docker.exception.SourceNotFoundException;
+import org.eclipse.che.workspace.infrastructure.docker.model.DockerContainerConfig;
+import org.eclipse.che.workspace.infrastructure.docker.monit.DockerMachineStopDetector;
+import org.eclipse.che.workspace.infrastructure.docker.strategy.ServerEvaluationStrategyProvider;
+import org.slf4j.Logger;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+import static org.eclipse.che.workspace.infrastructure.docker.DockerMachine.CHE_WORKSPACE_ID;
+import static org.eclipse.che.workspace.infrastructure.docker.DockerMachine.LATEST_TAG;
+import static org.eclipse.che.workspace.infrastructure.docker.DockerMachine.USER_TOKEN;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * @author Alexander Garagatyi
+ */
+public class DockerMachineStarter {
+    private static final Logger LOG                     = getLogger(DockerMachineStarter.class);
+    /**
+     * Prefix of image repository, used to identify that the image is a machine saved to snapshot.
+     */
+    public static final  String MACHINE_SNAPSHOT_PREFIX = "machine_snapshot_";
+
+    public static final Pattern SNAPSHOT_LOCATION_PATTERN = Pattern.compile("(.+/)?" + MACHINE_SNAPSHOT_PREFIX + ".+");
+
+    static final   String            CONTAINER_EXITED_ERROR = "We detected that a machine exited unexpectedly. " +
```

Comment:
Strange formatting here, please check that it is OK. Also maybe it can be private.